Engine/CMakeLists.txt

```diff
@@ -3,11 +3,9 @@ cmake_minimum_required(VERSION 3.4)
 project(MoltenTempest)
 set(CMAKE_CXX_STANDARD 14)
 
+option(BUILD_SHARED_LIBS "Build shared libraries." OFF)
```

Comment:
Better to name Tempest properties with TEMPEST_ prefix, to avoid name collision.
Also, lets put shared as ON by default

Reply:
BUILD_SHARED_LIBS is the standard variable for exactly this purpose and it gets propagated to subprojects so for example when you enable BUILD_SHARED_LIBS in OpenGothic it automatically builds MoltenTempest as shared library.

Reply:
Automatically for everything can be very painful, if you have mixed project.
One library can be GPL(and you want shared for those), and other library - MIT(can be static).
From engine side, better to give a precise option to developer.

Reply:
I still don't get what you're talking about. The patch make it possible to build it as either shared or static (cmake dir -DBUILD_SHARED_LIBS=On/Off).

Reply:
Name collision is not a desirable behavior for engine - just use prefix.

Reply:
It's not a name collision, I'm making it possible to set the internal variable BUILD_SHARED_LIBS. It's literally recommended by the cmake docs (https://cmake.org/cmake/help/v3.4/variable/BUILD_SHARED_LIBS.html?highlight=build_shared_libs).

Reply:
I don't know, why cmake authors recommend such, but no.
In case of OpenGothic it's luse-luse case:
  All static in debug is to large - 200mb in debug
  All shared - generates tonn of *.dll with unused code, like squish.dll, libBulletSoftBody.dll

Reply:
We can introduce more options in OpenGothic then to control which dependencies to build statically and dynamically but you should not hardcode all MoltenTempest builds to be static.

Reply:
Can we meet in the middle  and implement solution, like this:
```
if (BUILD_SHARED_LIBS)
  set(TEMPEST_BUILD_SHARED_LIBS_DEFAULT ON)
else (BUILD_SHARED_LIBS)
  set(TEMPEST_BUILD_SHARED_LIBS_DEFAULT OFF)
endif (BUILD_SHARED_LIBS)
option(TEMPEST_BUILD_SHARED_LIBS "Build Shared Libraries" ${TEMPEST_BUILD_SHARED_LIBS_DEFAULT })
```
?
It fits my requirement for dedicated option and doesn't affect your use case with `BUILD_SHARED_LIBS`

Reply:
It's unnecessary. We can set the BUILD_SHARED_LIBS variable on a per dependency basis in OpenGothic. I've updated the pull request in OpenGothic and you'd just set -DBUILD_SHARED_MOLTEN_TEMPEST=On when building OpenGothic to get the behavior you want. No changes to this pull request required.